### DISCUSSION TITLE
Come ottimizzare delle automazioni con notifiche

### DISCUSSION THREAD

**Discussion by liuk4friends** [indomus/forum#2656] [HUB: Home Assistant]
Ciao a tutti,
Da giugno ho un impianto fotovoltaico con inverter a raffreddamento passivo.

Dopo un po' mi sono accorto che quando l'inverter superava i 60° di temperatura avevo un calo di produzione (in effetti, le specifiche dell'inverter indicano proprio quella temperatura come soglia massima)

Con tre ventole PC assemblate in serie (dotate di trasformatore e potenziometro, kit preso su AliExpress) ed un sonoff mini ho impostato un'automazione che fa partire le suddette ventole quando l'inverter supera i 42° nonché un'altra automazione che le spegne quando la temperatura scende sotto i 38°.

Entrambe le automazioni mi notificano via telegram l'accensione o lo spegnimento delle ventole.

Tutto funziona perfettamente (le piccole soddisfazioni del principiante) ma ho un unico piccolo problema.

Nel corso della giornata mi arrivano decine di notifiche perché ovviamente la temperatura scende quando partono le ventole e si rialza dopo 10/15 minuti. Questo determina notifiche continue.

Mi sto scervellando per capire come potrei avere una sola notifica quando - la prima volta nella giornata - HA fa partire le ventole e una sola notifica l'ultima volta nella giornata che la temperatura scende sotto i 38 e le ventole si spengono sino al giorno dopo.

Quale settaggio oppure quale condizione potrei inserire nelle automazioni per ottenere questo risultato?
grazie a tutti per l'attenzione.

Luca

**Comment by 50494554524F** [comment]
usa una logbook card se vuoi vederlo nell'interfaccia oppure dei sensori history_stats di tipo count e con un'automazione ogni tot ore ti invii il valore del contatore... più logico... anche perché l'ultima notifica immagino dovrai inviartela dopo il tramonto, se proprio vuoi fare così usa degli input_datetime (ad esempio) per salvarti la prima accensione e gli spegnimenti, poi decidi quando inviarti la notifica sfruttando l'integrazione sun

**Comment by liuk4friends** [comment]
Innanzitutto grazie della risposta.
Mi hai dato un ottimo suggerimento.

Per quanto riguarda la prima notifica di avvio delle ventole, proverò a creare un sensore history_stats di tipo "count" basato sullo stato del sonoff mini (che appunto accende e spegne le ventole), inserendo nell'automazione una condizione per cui questo specifico sensore non deve essere superiore a 1.
Così facendo dovrebbe arrivarmi la notifica relativa alla prima volta in cui le ventole partono e poi non dovrebbero arrivarmene più.

In relazione invece alla notifica di spegnimento delle ventole, per fare in modo che me ne arrivi una sola in occasione dell'ultimo spegnimento, solitamente nel tardo pomeriggio, potrei inserire una condizione basata proprio sul tempo. 
Poichè, infatti, nel corso della giornata il ciclo di accensione e spegnimento delle ventole è di circa 10/15 minuti, potrei inserire la condizione che la temperatura deve essere inferiore a 38 ° da più di 30 minuti.

Stasera provo a inserire queste modifiche e domani vedo come va,

**Comment by liuk4friends** [comment]
Test di oggi concluso in modo positivo.

La prima notifica (relativa al primo momento della giornata in cui partono le ventole per raffreddare l'inverter) l'ho risolta come avevo immaginato nel post precedente.

Per la seconda notifica (relativa all'ultimo spegnimento delle ventole nella giornata in corso) ho proceduto in modo diverso.
Avendo già una notifica al tramonto che mi segnala il livello di carica dell'accumulo in quel momento, ho allungato il testo con il numero di volte in cui le ventole sono state accese e per quanto tempo hanno funzionato nel corso della giornata.

Lascio aperto il thread per qualche giorno nel caso in cui qualcuno avesse soluzioni migliori. Poi lo chiuderò.
Buona serata a tutti
